src/content/en/fundamentals/getting-started/primers/async-functions.md

```diff
@@ -0,0 +1,249 @@
+
+          const value = result.value;
+          total += value.length;
+          console.log('Received chunk', value);
+
+          return reader.read().then(processResult);
+        })
+      });
+    }
+
+Look at that! See how I'm calling `processResult` inside itself to set up an asynchronous loop? Writing that made me feel *very smart*. But like most "smart" code you have to stare at it for ages to figure out what it's doing, like one of those magic-eye pictures from the 90's.
+
+Let's try that again with async functions:
+
+    async function getResponseSize(url) {
+      const response = await fetch(url);
+      const reader = response.body.getReader();
+      let result;
+      let total = 0;
+
+      while ((result = await reader.read()) && !result.done) {
+        const value = result.value;
+        total += value.length;
+        console.log('Received chunk', value);
+      }
+
+      return total;
+    }
+
+The asynchronous loop that made me feel so smug is replaced with a trusty, boring, while-loop. Much better.
+
+Note: The phrase "logging out the chunks" made me sick in my mouth.
+
+## Example: Avoid going too linear
+
+Ok, last example. Say we wanted to fetch a sequence of URLs and log them as soon as possible, in the correct order.
+
+*Deep breath* - here's how that looks with promises:
+
+    function logInOrder(urls) {
+      // go fetch
+      const textPromises = urls.map(url => {
+        return fetch(url).then(response => response.text());
+      });
+
+      textPromises.reduce((chain, textPromise) => {
+        return chain.then(() => textPromise)
+          .then(text => console.log(text));
+      }, Promise.resolve());
+    }
+
+Check me out, Jake "wielder of promises" Archibald, using `reduce` to process a sequence of promises. That's definitely a bit of *so smart* coding we'd be better off without.
+
+When converting the above to an async function, it's tempting to go *too linear*:
+
+<span class="compare-worse">Not recommended</span> - too linear
+
+    async function logInOrder(urls) {
+      for (const url of urls) {
+        const response = await fetch(url);
+        console.log(await response.text());
+      }
+    }
+
+Looks much neater, but my second fetch doesn't begin until my first fetch has been fully read, and so on. This is much slower than the promises example that performs the fetches in parallel. Thankfully there's an ideal middle-ground:
+
+<span class="compare-better">Recommended</span> - nice and parallel
+
+    async function logInOrder(urls) {
+      // go fetch
+      const textPromises = urls.map(async url => {
+        const response = await fetch(url);
+        return response.text();
+      });
+
+      for (const textPromise of textPromises) {
+        console.log(await textPromise);
+      }
+    }
+
+In this example, the URLs are fetched and read as text in parallel, but the "smart" `reduce` bit is replaced with a standard, readable, for-loop.
+
+## Browser support & workarounds
+
```

Comment:
What do you think about using a framebox here to embed the chromestatus info:
https://www.chromestatus.com/feature/5643236399906816?embed